### PR TITLE
Feature/doc builder v4

### PR DESCRIPTION
**Pull request by BrennanGit**
do not merge until https://github.com/xmos/doc_builder/pull/67 is released and we've updated the image tag

### CODE REVIEW

**Comment by BrennanGit** [comment]
examples found here: http://srv-bri-jcim0:8080/job/XMOS/job/lib_xcore_math/job/PR-154/lastBuild/artifact/doc/_build/

File: Jenkinsfile [hunk 1]
```diff
@@ -101,23 +101,18 @@ pipeline {
...
-                        -e DOXYGEN=1 -e DOXYGEN_INCLUDE=/build/doc/Doxyfile.inc \
-                        -e PDF=1 \
-                        ghcr.io/xmos/doc_builder:${XMOSDOC_VERSION} \
-                        || echo "PDF build is badly broken, ignoring for now till it's fixed." """
-                
+                sh "docker pull ghcr.io/xmos/doc_builder:$XMOSDOC_VERSION"
+                sh """docker run -u "\$(id -u):\$(id -g)" \
+                    --rm \
+                    -v ${WORKSPACE}:/build \
+                    ghcr.io/xmos/doc_builder:$XMOSDOC_VERSION -v"""
```
Comment (lucianomartin) [inline Jenkinsfile:115]: can we zip the doc files so that the artifacts look tidier?

Comment (BrennanGit) [reply Jenkinsfile:115]: We could but then we lose the ability to render them out directly, you'd have to download and open them like with github actions. (eg http://srv-bri-jcim0:8080/job/XMOS/job/lib_xcore_math/job/PR-154/2/artifact/doc/_build/html/index.html)

If you specifically want them in a zip, classic mode has a button for that.

http://srv-bri-jcim0:8080/job/XMOS/job/lib_xcore_math/job/PR-154/2/artifact/doc/_build/

image

Comment (lucianomartin) [reply Jenkinsfile:115]: Thanks for the info, but the Blue Ocean artifacts overview, which most of the developers use, is pretty useless due to the large number of files.

File: Jenkinsfile [hunk 2]
```diff
@@ -101,23 +101,18 @@ pipeline {
...
               steps {
                 runningOn(env.NODE_NAME)
                 checkout scm
-                sh "docker pull ghcr.io/xmos/doc_builder:${XMOSDOC_VERSION}"
-                sh """docker run --user "\$(id -u):\$(id -g)" \
-                        --rm \
-                        -v ${WORKSPACE}:/build \
-                        -e EXCLUDE_PATTERNS="/build/doc/doc_excludes.txt" \
-                        -e DOXYGEN=1 -e DOXYGEN_INCLUDE=/build/doc/Doxyfile.inc \
-                        -e PDF=1 \
```
Comment (lucianomartin) [inline Jenkinsfile:117]: this line has been removed, but the PDF are still generated. How do we disable the PDP generation with `xmosdoc`?

File: Jenkinsfile (hunk 1 above)
Comment (BrennanGit) [reply Jenkinsfile]: I guess there's a discussion to be had which is more valuable.

Theres a blue link at the very top of that page called "Build Artifacts".

image

File: Jenkinsfile (hunk 2 above)
Comment (BrennanGit) [reply Jenkinsfile:117]: The default builders for `xmosdoc` are `clean linkcheck html latex`. 
To run different builders you just choose that: `xmosdoc clean linkcheck html` (or just `xmosdoc html`)

**Pull request closed without merging**